source/common/stream_info/stream_info_impl.h

```diff
@@ -131,6 +131,13 @@ struct StreamInfoImpl : public StreamInfo {
     final_time_ = time_source_.monotonicTime();
   }
 
+  DownstreamTiming& downstreamTiming() override {
+    if (!downstream_timing_.has_value()) {
```

Comment:
Why have `downstream_timing_` as optional? Seems like we could just have the struct if we are constructing on absence?

Reply:
I could go either way here, but I don't think we set these for TCP, and TCP (so redis etc.) have stream info, so I was largely trying to spare them the memory footprint.

Reply:
ah gotcha!